Mage/src/main/java/mage/abilities/AbilityImpl.java

```diff
@@ -558,7 +558,11 @@ protected VariableManaCost handleManaXCosts(Game game, boolean noMana, Player co
                     } else {
                         String manaSymbol = null;
                         if (variableManaCost.getFilter().isBlack()) {
-                            manaSymbol = "B";
+                            if (variableManaCost.getFilter().isRed()) {
+                                manaSymbol = "B/R";
```

Comment:
Yes, these two lines above are required. The part that may be hacky is leveraging the hybrid mana functionality for the Soul Burn card. It came out before that existed. The functionality described on the card is exactly satisfied by this solution though. 

I don't believe there's any reason to ever implement all the other multi mana possibilities, and therefore have a ton of if statements here; since no card would ever leverage them. It makes my OCD twitch a bit at not being thorough/complete on something, but it seems absurd to write a bunch of code that nothing will ever exercise.